src/tracks/Pattern.cpp

```diff
@@ -453,6 +461,48 @@ void Pattern::loadSettings( const QDomElement & _this )
 		m_steps = MidiTime::stepsPerTact();
 	}
 
+	// Backward compatibility for < 1.2.0
+	if( ! _this.hasAttribute( "version" ) )
+	{
+		if( m_patternType == BeatPattern )
+		{
+			MidiTime oldLength = MidiTime( _this.attribute( "len" ).toInt() );
+
+			tick_t max_length = MidiTime::ticksPerTact();
+			int oldSteps = m_steps;
```

Comment:
Thanks, but if you follow the thread you will see that this is old code that's going to change completely. Also, I use the tag `In progress` for stuff that isn't really ready for review.